client/cmd/dexc/config.go

```diff
@@ -1,231 +1,38 @@
-	SiteDir      string `long:"sitedir" description:"Path to the 'site' directory with packaged web files. Only used with --no-embed-site. When unspecified, search known locations relative to dexc."`
-	DebugLevel   string `long:"log" description:"Logging level {trace, debug, info, warn, error, critical}"`
-	LocalLogs    bool   `long:"loglocal" description:"Use local time zone time stamps in log entries."`
-	CPUProfile   string `long:"cpuprofile" description:"File for CPU profiling."`
-	HTTPProfile  bool   `long:"httpprof" description:"Start HTTP profiler on /pprof."`
-	ShowVer      bool   `short:"V" long:"version" description:"Display version information and exit"`
-	TorProxy     string `long:"torproxy" description:"Connect via TOR (eg. 127.0.0.1:9050)."`
-	TorIsolation bool   `long:"torisolation" description:"Enable TOR circuit isolation."`
-	Onion        string `long:"onion" description:"Proxy for .onion addresses, if torproxy not set (eg. 127.0.0.1:9050)."`
-	Net          dex.Network
-	CertHosts    []string
-	Experimental bool `long:"experimental" description:"Enable experimental features"`
-
-	NoAutoWalletLock   bool `long:"no-wallet-lock" description:"Disable locking of wallets on shutdown or logout. Use this if you want your external wallets to stay unlocked after closing the DEX app."`
-	NoAutoDBBackup     bool `long:"no-db-backup" description:"Disable creation of a database backup on shutdown."`
-	UnlockCoinsOnLogin bool `long:"release-wallet-coins" description:"On login or wallet creation, instruct the wallet to release any coins that it may have locked."`
-}
-
-var defaultConfig = Config{
-	AppData:    defaultApplicationDirectory,
-	Config:     defaultConfigPath,
-	DebugLevel: defaultLogLevel,
-	CertHosts: []string{defaultTestnetHost, defaultSimnetHost,
-		defaultMainnetHost},
-}
-
-// configure processes the application configuration.
-func configure() (*Config, error) {
+func configure() (*app.Config, error) {
 	// Pre-parse the command line options to see if an alternative config file
 	// or the version flag was specified. Override any environment variables
 	// with parsed command line flags.
-	iniCfg := defaultConfig
+	iniCfg := app.DefaultConfig
 	preCfg := iniCfg
-	preParser := flags.NewParser(&preCfg, flags.HelpFlag|flags.PassDoubleDash)
-	_, flagerr := preParser.Parse()
-
-	if flagerr != nil {
-		e, ok := flagerr.(*flags.Error)
-		if !ok || e.Type != flags.ErrHelp {
-			preParser.WriteHelp(os.Stderr)
-		}
-		if ok && e.Type == flags.ErrHelp {
-			preParser.WriteHelp(os.Stdout)
-			os.Exit(0)
-		}
-		return nil, flagerr
+	if err := app.ParseCLIConfig(&preCfg); err != nil {
+		return nil, err
 	}
 
 	// Show the version and exit if the version flag was specified.
 	if preCfg.ShowVer {
-		fmt.Printf("%s version %s (Go version %s %s/%s)\n",
-			appName, Version, runtime.Version(), runtime.GOOS, runtime.GOARCH)
+		fmt.Printf("App version %s (Go version %s %s/%s)\n",
```

Comment:
Why no `appName`?  It's saying "dexc" in main: https://github.com/decred/dcrdex/pull/1957/files#diff-090ed4c3ba3c62df58714d954e34f7179d481afdfa15ef34aa3d195207f6dbb0R77

Over in dexc-desktop, I also would not object to more than just "Desktop app version..."  I think it should identify itself there too, although adding the "desktop" qualifier seems great.